### ISSUE TITLE
Rules constructor-super and no-this-before-super are applying for non derived class

### ISSUE THREAD

**Issue by unlight** [eslint/typescript-eslint-parser#39]
```
  14:5  error  Expected to call 'super()'              constructor-super
  15:9  error  'this' is not allowed before 'super()'  no-this-before-super
```
```js
export class MockRouteSegment implements RouteSegment {
    urlSegments: any;
    parameters: any;
    outlet: string;
    _type: any;
    _componentFactory: any;
    type: any;
    stringifiedUrlSegments: string;

    constructor(parameters?: { [key: string]: any; }) { // L:14
        this.parameters = parameters; // L:15
    }

    getParam(param: string) {
        return this.parameters[param];
    }
}
```
`MockRouteSegment` is not derived class.

**Comment by nzakas** [comment]
ESLint rules are not expected to work with this parser yet.

**Comment by nzakas** [comment]
Reopening this as I realized that there is an error in the way class `implements` is converted vs. class `extends`. In TypeScript they are represented more or less the same in the AST whereas we need to separate them.
